src/stdio_mgr/stdio_mgr.py

```diff
@@ -72,15 +72,17 @@ class TeeStdin(StringIO):
 
     tee = attr.ib(validator=attr.validators.instance_of(TextIOBase))
     init_text = attr.ib(default="", validator=attr.validators.instance_of(str))
+    _encoding = attr.ib(default="utf-8", validator=attr.validators.instance_of(str))
 
     def __attrs_post_init__(self):
         """Call normal __init__ on superclass."""
-        super().__init__(self.init_text)
+        self._buf = BytesIO(self.init_text.encode(self._encoding))
```

Comment:
Should this be `self.buffer`? Or ... does `TextIOWrapper` retrieve `self._buf` when a user tries to access `self.buffer`?

OHHH. `self.buffer` is okay when *accessing* the buffer... but for this *definition* of the buffer, setting `self.buffer` would override the `TextIOWrapper.buffer` machinery?

DOUBLE-OH. `self.buffer` isn't even available yet, because `super().__init__` hasn't been called yet!

TRIPLE-OH. `self._buf` is the `BytesIO`, whereas `self.buffer` ends up pointing to the `BufferedReader` enclosing the `BytesIO`. So, `self.buffer` is read-only, but `self._buf` is read-write.